src/main/java/org/opentripplanner/routing/core/ItineraryFares.java

```diff
@@ -1,42 +1,64 @@
 package org.opentripplanner.routing.core;
 
-import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.LinkedHashMultimap;
 import com.google.common.collect.Multimap;
 import java.util.Collection;
 import java.util.HashMap;
-import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
-import org.opentripplanner.ext.fares.model.FareProduct;
+import javax.annotation.Nullable;
 import org.opentripplanner.ext.fares.model.LegProducts;
 import org.opentripplanner.framework.tostring.ToStringBuilder;
+import org.opentripplanner.model.fare.FareProduct;
+import org.opentripplanner.model.fare.FareProductUse;
 import org.opentripplanner.model.plan.Leg;
 import org.opentripplanner.transit.model.basic.Money;
 
 /**
- * <p>
- * ItineraryFares is a set of fares for different classes of users.
- * </p>
+ * ItineraryFares is a set of fares for different legs, rider categories or fare media.
  */
 public class ItineraryFares {
```

Comment:
Could this class be moved to the fare package?

Reply:
I moved it and all of its sandbox dependencies in 27a489483d10f7b6548ff74ac5e54403d91c0920

I acknowledge that there is room for improvement there: the relationships can be untangled better and some of them can go back into the sandbox. I would like to do that in a follow up PR.